src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
> I prefer CreateA and CreateB because they give an indication on when each should be used (I'm the same way on overloads generally, I prefer to have different names indicating when to use each if they are really for different purposes)

My main concern here is following existing patterns in the code base (personal pref is CreateA, CreateB methods).  I tried searching through the code base for guidance and unfortunately there isn't a lot to be had.  

- Types which have CreateA / CreateB factory model: 1
- Types which have Create only: 0

There are more examples for `internal` types but for `public` types there is very little data (or my program to look for it had a bug).  

> I think the new public API here should have a name which discourages actually using it, because we don't want people to use it.

I'm not opposed to this, but I think using `Serialized` is the wrong terminology.  It's implying a very specific use which we are currently removing from the compiler.